src/braket/tasks/local_quantum_task.py

```diff
@@ -42,8 +42,12 @@ def id(self) -> str:
         return str(self._id)
 
     def cancel(self) -> None:
-        """Cancel the quantum task."""
-        raise NotImplementedError("Cannot cancel completed local task")
+        """Attempt to cancel the quantum task.
+
+        Since this class is instantiated with the results, there is nothing to cancel. Attempting
+        to cancel an already completed task is not an error.
+        """
+        pass
```

Comment:
I have mixed feelings about using `pass` here. What would your thoughts be if we instead used a log line indicating local tasks are already complete on instantiation or some better worded statement?